client/components/cart-summary.jsx

```diff
@@ -4,14 +4,24 @@ import priceFormatter from '../price-format';
 
 export default function CartSummary(props) {
   const totalPrice = props.cartState.reduce((accumulator, currentValue) => accumulator += currentValue.price, 0);
+  function buttonRender() {
+    if (props.cartState.length) {
+      return <button onClick={() => props.setView('checkout', {})} className="btn btn-primary">Checkout</button>;
+    } else {
+      return null;
+    }
+  }
   return (
     <div className="row">
       <button onClick={() => props.setView('catalog', {})}className="btn">&#8592; Back to Catalog</button>
       <h1 className="col-12 mt-3 mb-4">My Cart</h1>
       <div className="row">
         {props.cartState.map(item => <CartSummaryItem key={item.productId} cartData={item}/>)}
       </div>
-      <h2 className="mt-3">Item Total <span>{priceFormatter(totalPrice)}</span></h2>
+      <div className=" col-12 d-flex justify-content-between pb-5">
+        <h2 className="mt-3">Item Total <span>{priceFormatter(totalPrice)}</span></h2>
+        {buttonRender()}
```

Comment:
I didn't realize that you were talking about a function to render something. Instead of `buttonRender()` you can inline the conditional rendering with `&&`.

```jsx
{ props.cartState.length !== 0 &&
  <button onClick={() => props.setView('checkout', {})} className="btn btn-primary">Checkout</button>
}
```

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator